scipy/io/matlab/mio5.py

```diff
@@ -762,6 +786,28 @@ def write_object(self, arr):
         self._write_items(arr)
 
 
+class ZlibOutputStream(object):
+    ''' Simple container for writing to a zlib compression stream '''
+    def __init__(self, file_stream):
+        self.file_stream = file_stream
+        self.zobj = zlib.compressobj()
+
+    def write(self, filestr):
+        if len(filestr) < (2**31):
+            self.file_stream.write(self.zobj.compress(filestr))
+        else:
+            rounds = int(np.ceil(1.0*len(filestr)/(2**30)))
+            for idx in xrange(rounds):
+                substr = filestr[idx * (2**30):(idx+1) * (2**30)]
+                self.file_stream.write(self.zobj.compress(substr))
```

Comment:
this could be simplified to:

```python
for idx in range(0, len(filestr), 2**30):
    self.file_stream.write(self.zobj.compress(filestr[idx:idx + 2**30]))
```